src/modules/error-catalog.ts

```
@@ -0,0 +1,79 @@
+    const catalogEntry = ErrorCatalog.catalog[code];
+
+    if (!catalogEntry) {
+      throw new Error(`Error code ${code} not found in catalog`);
+    }
+
+    // Clone the entry to avoid mutating the original catalog entry
+    const entryWithFormattedMessage = { ...catalogEntry };
+
+    // Format the message with metadata if provided
+    if (metadata) {
+      entryWithFormattedMessage.message = this.formatMessage(catalogEntry.message, metadata);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)